credmark/model/ledger/ledger.py

```diff
@@ -144,3 +149,32 @@ def get_erc20_transfers(self, columns: List[str],
                                      TokenTransferTable,
                                      columns, where, group_by,
                                      order_by, limit, offset)
+
+    def get_block_number_of_date(self, dt_in: Union[datetime, date]):
+        txn_blocks = self.Block.Columns
+        dt = datetime(dt_in.year, dt_in.month, dt_in.day, tzinfo=timezone.utc)
+        res = self.get_blocks(
+            [txn_blocks.TIMESTAMP, txn_blocks.NUMBER],
+            where=f"{txn_blocks.TIMESTAMP} >= {int(dt.timestamp())}",
+            order_by=f'{txn_blocks.TIMESTAMP} ASC',
+            limit='1')
+
+        rows = res.data
+        block_hist = rows[0].get(txn_blocks.NUMBER) if len(rows) else None
+
+        return block_hist
+
+    def get_date_of_block_number(self, block_hist: int):
+        txn_blocks = self.Block.Columns
+        res = self.get_blocks(
```

Comment:
use model rpc.get-block-timestamp in model-runner which returns {"timestamp": 123}